app/models/user.rb

```diff
@@ -412,6 +412,10 @@ def provider_id_for_audits
     account.try!(:provider_id_for_audits) || provider_account.try!(:provider_id_for_audits)
   end
 
+  def requires_strong_password?
```

Comment:
What's the purpose of adding this extra method?.. :thinking: 
Can't you just use `provider_requires_strong_passwords?` in validations?

Reply:
I need to add this extra method because u is a decorator object and defined method is private method and so I have call this private method from the method defined in class. the issue is with this method is of undefined only.


but yes I agree we can directly also use this method so code is updated

Reply:
Rather than adding this method, I would just move `def provider_requires_strong_passwords?` out of private methods.

Reply:
Okay if we could remove from that private then it can be done, I was just thinking to not remove from private as it a password related method but then it can be used in validation as you mentioned below.

changes updated